### PR TITLE
docker engine and compose version check

### PR DESCRIPTION
**Pull request by estenberg**
Some users got unexpected behavior due to too old version of docker and docker-compose.
Minimum required is aligned with docs.

### CODE REVIEW

**Comment by estenberg** [comment]
@bboozzoo could you review pls?

File: verify-docker-versions [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  echo "DOCKER_ENGINE_VERSION_STRING=$DOCKER_ENGINE_VERSION_STRING"
+  exit 1
+fi
+
+
+# test Docker Compose version
+
+DOCKER_COMPOSE_VERSION_REGEX_MINOR_BACKREF="docker-compose version 1\.([0-9]+)\.[0-9]+, build [0-9a-f]+"
+
+DOCKER_COMPOSE_VERSION_STRING=$(docker-compose version 2>/dev/null | head -n1)
```
Comment (bboozzoo) [inline verify-docker-versions]: I think it's better to try `docker-compose version --short`:

```
mborzecki:mendersoftware/deployments/tests$ docker-compose version 
docker-compose version 1.10.0, build 4bd6f1a
docker-py version: 2.0.2
CPython version: 2.7.13
OpenSSL version: OpenSSL 1.0.2k-fips  26 Jan 2017
mborzecki:mendersoftware/deployments/tests$ docker-compose version --short
1.10.0
```

File: verify-docker-versions [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  return 0
+}
+
+
+# test Docker Engine version
+
+DOCKER_ENGINE_VERSION_REGEX_MINOR_BACKREF="Docker version 1\.([0-9]+)\.[0-9]+, build [0-9a-f]+"
+
+DOCKER_ENGINE_VERSION_STRING=$(docker --version 2>/dev/null | head -n1)
```
Comment (bboozzoo) [inline verify-docker-versions]: I would replace this with `docker info`:

```
mborzecki:mendersoftware/deployments/tests$ docker info -f '{{.ServerVersion}}'
1.13.1
```

`docker info` pings docker daemon. AFAIK it's possible to have a different `docker` version A locally and talk to a remote daemon in version B. Another advantage is that if someone is using `DOCKER_HOST` we'll ping the right daemon.

**Comment by estenberg** [comment]
Thanks for the feedback @bboozzoo - let me know if this looks ok to merge.

**Pull request merged**